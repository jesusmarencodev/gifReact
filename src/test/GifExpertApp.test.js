import React from 'react';
import '@testing-library/jest-dom'; //autocompletado mejorado
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Prueba componente <GifExpertApp />', () => {
	const categories = ['Dragon Ball', 'One punch'];

	test('Debe cargar correctamente el componente', () => {
		const wrapper = shallow(<GifExpertApp/>);
		expect(wrapper).toMatchSnapshot();
	});


	test('Debe de mostrar una lista de categorias', () => {
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});

	test('Debe de existir el componente <AddCategory />', () => {
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
		expect(wrapper.find('AddCategory').exists()).toBe(true);
	})
	
	
});
