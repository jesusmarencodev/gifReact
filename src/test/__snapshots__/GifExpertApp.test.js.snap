// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente <GifExpertApp /> Debe cargar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Prueba componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
  <GifGrid
    category="One punch"
    key="One punch"
  />
</Fragment>
`;
